packages/monitors/index.js

```diff
@@ -80,229 +88,281 @@ async function run({
 
     const getTime = () => Math.round(new Date().getTime() / 1000);
 
-    const [detectedContract, networkId, latestBlock] = await Promise.all([
-      findContractVersion(financialContractAddress, web3),
-      web3.eth.net.getId(),
-      web3.eth.getBlock("latest")
-    ]);
-
-    // Append the contract version and type to the monitorConfig, if the monitorConfig does not already contain one.
-    if (!monitorConfig) monitorConfig = {};
-    if (!monitorConfig.contractVersion) monitorConfig.contractVersion = detectedContract.contractVersion;
-    if (!monitorConfig.contractType) monitorConfig.contractType = detectedContract.contractType;
-
-    // Check that the version and type is supported. Note if either is null this check will also catch it.
-    if (
-      SUPPORTED_CONTRACT_VERSIONS.filter(
-        vo => vo.contractType == monitorConfig.contractType && vo.contractVersion == monitorConfig.contractVersion
-      ).length == 0
-    )
-      throw new Error(
-        `Contract version specified or inferred is not supported by this bot. Monitor config:${JSON.stringify(
-          monitorConfig
-        )} & detectedContractVersion:${JSON.stringify(detectedContract)} is not part of ${JSON.stringify(
-          SUPPORTED_CONTRACT_VERSIONS
-        )}`
+    // `asyncLoopLogic` is an async method that is called once for each loop iteration and will be retried
+    // specified.
+    let asyncLoopLogic;
+    /** *************************************
```

Comment:
This script shows a lot of diff but really the code is only reorganized. All of the existing `index.js` code is put into the `if(type==="financial-contract")` branch since its irrelevant for running the `type===optimistic-oracle` branch.